File: Cpp/CXR_classify/downloadstage.cpp
#include "downloadstage.h"


DownloadStage::DownloadStage(ConfigHandler * configHandler) : Stage1()
{
    DownloadStage::downloader = new Downloader1(configHandler);
}

void DownloadStage::download()
{
    threadpool->start(downloader);

}

File: Go/CxrClassify/downloadStage/downloadStage.go
package downloadStage

import (
	"CxrClassify/configHandler"
	"CxrClassify/databaseHandler"
	"CxrClassify/stage"
)

type DownloadStage struct {
	stage.Stage

	Downloader *Downloader

	_ func() `constructor:"init"`
}

func (d *DownloadStage) init() {

}

func (d *DownloadStage) Setup(configHandler *configHandler.ConfigHandler, databaseHandler *databaseHandler.DatabaseHandler) {
	d.Downloader = NewDownloader(nil)
	d.Downloader.Setup(configHandler, databaseHandler)
}

func (d *DownloadStage) Download() {
	d.Downloader.Run()
}
